metrics/analysis.go

```diff
@@ -0,0 +1,135 @@
+package metrics
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Default service metrics for analyzer operations.
+type AnalysisMetrics struct {
+	// Name of the runtime (or "consensus") that is being analyzed.
+	runtime string
+
+	// Counts of database operations
+	databaseOperations *prometheus.CounterVec
+
+	// Latencies of database operations.
+	databaseLatencies *prometheus.HistogramVec
+
+	// Cache hit rates for the local cache.
+	localCacheReads *prometheus.CounterVec
+
+	// Latencies of analysis.
+	blockAnalysisLatencies *prometheus.HistogramVec
+
+	// Latencies of fetching a block's data from the node.
+	blockFetchLatencies *prometheus.HistogramVec
+}
+
+type CacheReadStatus string
+
+const (
+	CacheReadStatusHit      CacheReadStatus = "hit"
+	CacheReadStatusMiss     CacheReadStatus = "miss"
+	CacheReadStatusBadValue CacheReadStatus = "bad_value" // Value in cache was not valid (likely because of mismatched types / CBOR encoding).
```

Comment:
oh nice do we not have that panic anymore when the value doesn't exist in the cache

Reply:
Discussed offline. If our memory serves, the panic was only triggered in very specific circumstances, which were made impossible in #555, with the removal of `query_on_cache_miss`.